drivers/shared/executor/resource_container_linux.go

```diff
@@ -23,3 +25,46 @@ func (rc *resourceContainerContext) executorCleanup() error {
 	}
 	return nil
 }
+
+func (rc *resourceContainerContext) isEmpty() bool {
+	return rc.groups == nil
+}
+
+func (e *UniversalExecutor) getAllPids() (map[int]*nomadPid, error) {
+	return e.resConCtx.getAllPids()
+}
+
+func (rc *resourceContainerContext) getAllPids() (map[int]*nomadPid, error) {
+	nPids := map[int]*nomadPid{}
+
+	if rc.groups == nil {
+		return nPids, nil
+	}
+
+	paths := map[string]string{}
+	if len(rc.groups.Paths) > 0 {
+		paths = rc.groups.Paths
+	} else {
+		paths[""] = rc.groups.Path
+	}
```

Comment:
For cgroups, we want to use a single cgroup hierarchy.  Checking multiple hierarchy would lead to duplicate values in the case of multiple cgroups attached.

In Nomad, we always create the `freezer` cgroup only in `configureBasicCgroups`[1], so that's an appropriate one to use.  For comparison, `docker/runc` requires `devices` cgroup to be mounted, so they only check the `devices` subsystem[2]

[1] https://github.com/hashicorp/nomad/blob/88c03d088850a07a505fff91ac3909cbfe71b656/drivers/shared/executor/executor_linux.go#L715-L728

[2] https://github.com/opencontainers/runc/blob/master/libcontainer/cgroups/fs/apply_raw.go#L286-L289

Reply:
in `configureBasicCgroups`, we make an empty `Freezer` and ask it it's name, which constantly returns "freezer". Should I do that here? or use just the string over there?